File: NipSearchApp/ClientApp/src/components/EntityPerson.js
import React, { Component } from 'react';

const EntityPerson = ({ data, title }) => {
    return (
        <>
            <h4 id="entityPerson">{title}</h4>
            <table className='table table-striped' aria-labelledby="tabelLabel">
                <thead>
                    <tr>
                        <th>Company Name</th>
                        <th>First Name</th>
                        <th>Last Name</th>
                        <th>Pesel</th>
                    </tr>
                </thead>
                <tbody>
                    {data.map(item => (
                        <tr key={item.nip}>
                            <td>{item.companyName}</td>
                            <td>{item.firstName}</td>
                            <td>{item.lastName}</td>
                            <td>{item.pesel}</td>
                        </tr>
                    ))}
                </tbody>
            </table>
        </>
    );
};

export default EntityPerson;

File: NipSearchApp/Models/EntityPerson.cs
namespace NipSearchApp.Models
{
    public class EntityPerson
    {
        public int id { get; set; }
        public string? companyName { get; set; }
        public string? firstName { get; set; }
        public string? lastName { get; set; }
        public string? pesel { get; set; }
        public string? nip { get; set; }

        public int subjectId { get; set; }
        public Subject subject { get; set; }
    }
}
